### PR TITLE
Copy response to client on failed handshake

### PR DESCRIPTION
**Pull request by jlhawn**

### CODE REVIEW

File: websocketproxy.go [hunk 1]
```diff
@@ -134,6 +135,14 @@ func (w *WebsocketProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	connBackend, resp, err := dialer.Dial(backendURL.String(), requestHeader)
 	if err != nil {
 		log.Printf("websocketproxy: couldn't dial to remote backend url %s\n", err)
+		if resp != nil {
```
Comment (janmiderback) [inline websocketproxy.go]: What if `resp` is nil? Could we then still write "Service Unavailable" or something appropriate to the `rw`?

Comment (jlhawn) [reply websocketproxy.go]: Yes, a `503 Service Unavailable` seems appropriate.

Comment (jlhawn) [reply websocketproxy.go]: Updated below this line with an `else { ... }` case. Please take a look.

Comment (janmiderback) [reply websocketproxy.go]: Nice. I think using:
```
http.Error(rw, "Service Unavailable", http.StatusServiceUnavailable)
```
goes more aligned with rest of the code. Also, whether
- `http.StatusServiceUnavailable` is the better, or
- `http.StatusInternalServerError`
should be considered by the repo owners.

Comment (rjeczalik) [reply websocketproxy.go]: I opt for 503 here.

@jlhawn  the PR looks good, thanks for the work. Can I ask you to update the code according to @janmiderback's suggestion?

```go
http.Error(rw, http.StatusText(http.StatusServiceUnavailable), http.StatusServiceUnavailable)
```

File: websocketproxy.go [hunk 2]
```diff
@@ -134,6 +135,16 @@ func (w *WebsocketProxy) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
...
+			// If the WebSocket handshake fails, ErrBadHandshake is returned
+			// along with a non-nil *http.Response so that callers can handle
+			// redirects, authentication, etcetera.
+			if err := copyResponse(rw, resp); err != nil {
+				log.Printf("websocketproxy: couldn't write response after failed remote backend handshake: %s\n", err)
```
Comment (rjeczalik) [inline websocketproxy.go]: nit: you don't need `\n` here, log.Printf does it for you, as oppose to fmt.Printf

**Comment by jlhawn** [comment]
@rjeczalik sorry for being late to update this but I just pushed a change which addresses all of the comments.

**Comment by rjeczalik** [comment]
Hey @jlhawn, no worries, thanks for the PR!

**Pull request merged**